Test Failure: com.ibm.ws.logstash.collector.tests.LogstashSSLTest.testLogstashForAuditEvent

RTC 279446

```
testLogstashForAuditEvent:junit.framework.AssertionFailedError: 2020-09-24-14:05:22:159 Did not find liberty_audit
at com.ibm.ws.logstash.collector.tests.LogstashSSLTest.testLogstashForAuditEvent(LogstashSSLTest.java:234)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:197)
at org.testcontainers.containers.FailureDetectingExternalResource$1.evaluate(FailureDetectingExternalResource.java:30)
at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:318)
at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:171)
```

Sometimes, the type `liberty_audit` is split into 2 lines in the logstash output so it failed the string matching.


```
[01/11/2021 00:22:35:236 PST] 085 LogstashCollectorTest          logstashContainer              I {"@timestamp":"2021-01-11T08:22:35.199Z","ibm_audit_observer.typeURI":"service/server","ibm_audit_target.typeURI":"service/application/web","ibm_audit_eventName":"SECURITY_AUTHN","sequence":"1610353353718_0000000000003","ibm_audit_reason.reasonType":"HTTP","ibm_audit_eventTime":"2021-01-11T00:22:33.710-0800","ibm_audit_target.params":"id=testLogstashForAuditEvent","ibm_audit_reason.reasonCode":"200","datetime":"2021-01-11T00:22:33.718-0800","threadId":"0000017d","@version":"1","ibm_audit_target.host.address":"127.0.0.1:8010","ibm_audit_eventSequenceNumber":"2","serverName":"LogstashServer","ibm_audit_target.appname":"TraceServlet","ibm_audit_target.session":"sSMU2g7Cbpj470NSCqfXRpo","ibm_audit_observer.name":"SecurityService","ibm_audit_initiator.host.agent":"Java/1.8.0_191","ibm_audit_initiator.host.address":"127.0.0.1","ibm_audit_observer.id":"websphere: ebcprh1632js5zdam-n.fyre.ibm.com:/home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/:LogstashServer","ibm_audit_target.method":"GET","type":"liberty
[01/11/2021 00:22:35:237 PST] 085 LogstashCollectorTest          logstashContainer              I _audit","ibm_audit_target.realm":"defaultRealm","wlpUserDir":"/home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/","tags":["beats_input_codec_plain_applied"],"ibm_audit_outcome":"success","ibm_audit_target.id":"websphere: ebcprh1632js5zdam-n.fyre.ibm.com:/home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/:LogstashServer","hostName":"ebcprh1632js5zdam-n.fyre.ibm.com","ibm_audit_target.name":"/LogstashApp/TraceURL"}
[01/11/2021 00:22:35:238 PST] 085 LogstashCollectorTest          logstashContainer              I {"@timestamp":"2021-01-11T08:22:35.200Z","ibm_audit_observer.typeURI":"service/server","ibm_audit_target.typeURI":"service/application/web","ibm_audit_eventName":"SECURITY_AUTHZ","sequence":"1610353353727_0000000000004","ibm_audit_reason.reasonType":"HTTP","ibm_audit_eventTime":"2021-01-11T00:22:33.727-0800","ibm_audit_target.params":"id=testLogstashForAuditEvent","ibm_audit_reason.reasonCode":"200","datetime":"2021-01-11T00:22:33.727-0800","threadId":"0000017d","@version":"1","ibm_audit_target.host.address":"127.0.0.1:8010","ibm_audit_eventSequenceNumber":"3","serverName":"LogstashServer","ibm_audit_target.appname":"TraceServlet","ibm_audit_target.session":"sSMU2g7Cbpj470NSCqfXRpo","ibm_audit_observer.name":"SecurityService","ibm_audit_initiator.host.agent":"Java/1.8.0_191","ibm_audit_initiator.host.address":"127.0.0.1","ibm_audit_observer.id":"websphere: ebcprh1632js5zdam-n.fyre.ibm.com:/home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/:LogstashServer","ibm_audit_target.method":"GET","type":"liberty
[01/11/2021 00:22:35:239 PST] 085 LogstashCollectorTest          logstashContainer              I _audit","ibm_audit_target.realm":"defaultRealm","wlpUserDir":"/home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/","tags":["beats_input_codec_plain_applied"],"ibm_audit_outcome":"success","ibm_audit_target.id":"websphere: ebcprh1632js5zdam-n.fyre.ibm.com:/home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/:LogstashServer","hostName":"ebcprh1632js5zdam-n.fyre.ibm.com","ibm_audit_target.name":"/LogstashApp/TraceURL"}
[01/11/2021 00:23:03:743 PST] 001 FATRunner                      evaluate                       S null
java.lang.AssertionError: Did not find liberty_audit
```

--- a/dev/com.ibm.ws.logstash.collector_fat/fat/src/com/ibm/ws/logstash/collector/tests/LogstashCollectorTest.java
+++ b/dev/com.ibm.ws.logstash.collector_fat/fat/src/com/ibm/ws/logstash/collector/tests/LogstashCollectorTest.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2018, 2020 IBM Corporation and others.
+ * Copyright (c) 2018, 2021 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
  * which accompanies this distribution, and is available at
@@ -67,7 +67,7 @@ public abstract class LogstashCollectorTest {
     protected abstract LibertyServer getServer();
 
     private static String APP_URL = null;
-
+    private static String buffer;
     private static CopyOnWriteArrayList<String> logstashOutput = new CopyOnWriteArrayList<String>();
 
     protected void setConfig(String conf) throws Exception {
@@ -194,11 +194,27 @@ public abstract class LogstashCollectorTest {
     // use to pipe container output to our standard FAT output logs (output.txt)
     private static void log(OutputFrame frame) {
         String msg = frame.getUtf8String();
-        if (msg.endsWith("\n"))
+        if (msg.endsWith("\n")) {
             msg = msg.substring(0, msg.length() - 1);
-        logstashOutput.add(msg);
+        }
 
-        Log.info(c, "logstashContainer", msg);
+        boolean isComplete = false;
+        if (msg.startsWith("{") && msg.endsWith("}")) {
+            buffer = msg;
+            isComplete = true;
+        } else if (msg.startsWith("{")) {
+            buffer = msg;
+        } else if (msg.endsWith("}")) {
+            buffer = buffer + msg;
+            isComplete = true;
+        } else {
+            buffer = buffer + msg;
+        }
+        if (isComplete) {
+            logstashOutput.add(buffer);
+            Log.info(c, "logstashContainer", buffer);
+            buffer = null;
+        }
     }
 
     protected static void clearContainerOutput() {

--- a/dev/com.ibm.ws.logstash.collector_fat/fat/src/com/ibm/ws/logstash/collector/tests/LogstashSSLTest.java
+++ b/dev/com.ibm.ws.logstash.collector_fat/fat/src/com/ibm/ws/logstash/collector/tests/LogstashSSLTest.java
@@ -421,7 +421,7 @@ public class LogstashSSLTest extends LogstashCollectorTest {
          * by checking 1. whether the operating system is Mac or linux 2. whether the machine is running IBM JDK
          * if both checks pass, this is the case
          **/
-        Log.info(c, methodName, "os_name: " + os.toLowerCase() + "\t java_jdk: " + System.getProperty("java.vendor"));
+        Log.info(c, methodName, "os_name: " + os + "\t java_jdk: " + System.getProperty("java.vendor"));
         String JAVA_HOME = System.getenv("JAVA_HOME");
         Log.info(c, methodName, "JAVA_HOME: " + JAVA_HOME);
         boolean healthCenterInstalled = false;
@@ -445,12 +445,12 @@ public class LogstashSSLTest extends LogstashCollectorTest {
                 Log.info(c, methodName, " unable to find heathcenter.jar, thus unable to produce gc events. Thus, this check will be by-passed");
             }
         }
-        if (os.toLowerCase().contains("mac") || !System.getProperty("java.vendor").toLowerCase().contains("ibm")
+        if (os.contains("mac") || !System.getProperty("java.vendor").toLowerCase().contains("ibm")
             || System.getProperty("java.vendor.url").toLowerCase().contains("sun") || !healthCenterInstalled) {
             return true;
         }
         // Skip zOS temporary as zOS JDK has a bug that does not return any GC information.
-        if (os.contains("OS/390") || os.contains("z/OS") || os.contains("zOS")) {
+        if (os.contains("os/390") || os.contains("z/os") || os.contains("zos")) {
             return true;
         }
         return false;

